src/ddmd/backend/cgcs.c

```diff
@@ -415,7 +415,8 @@ STATIC void ecom(elem **pe)
             dbg_printf("i: %2d Hhash: %6d Helem: %p\n",
                 i,hcstab[i].Hhash,hcstab[i].Helem);
 #endif
-        if (hash == hcstab[i].Hhash && (ehash = hcstab[i].Helem) != NULL)
+        typeof(hcstab[0]) entry = hcstab[i];
+        if (hash == entry.Hhash && (ehash = entry.Helem) != NULL)
```

Comment:
I'd check the generated code and see if it is an actual improvement. I doubt it, as compilers do these sorts of optimizations automatically.